pkg/nodeagent/store/factory/storefactory.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package factory
+
+import (
+	"fmt"
+
+	"centaurusinfra.io/fornax-serverless/pkg/nodeagent/store"
+	"centaurusinfra.io/fornax-serverless/pkg/nodeagent/store/sqlite"
+)
+
+type PodStore struct {
+	store.Store
+}
+
+type SessionStore struct {
+	store.Store
+}
+
+type ContainerStore struct {
+	store.Store
+}
+
+func NewPodSqliteStore(options *sqlite.SQLiteStoreOptions) (*PodStore, error) {
+	if store, err := sqlite.NewSqliteStore("Pod", options,
+		func(text string) (interface{}, error) { return store.JsonToPod(text) },
+		func(obj interface{}) (string, error) { return store.JsonFromPod(obj.(*store.Pod)) }); err != nil {
+		return nil, err
+	} else {
+		return &PodStore{store}, nil
+	}
+}
+
+func (s *PodStore) GetPod(identifer string) (*store.Pod, error) {
+	obj, err := s.GetObject(identifer)
+	if err != nil {
```

Comment:
type conversion code seems repetitive in other places too. Is it possible to leverage golang generics?

Reply:
golang 1.18+ has generic; not mandatory to use it here; just in case it is useful.